src/compiler/checker.ts

```diff
@@ -13635,7 +13993,7 @@ namespace ts {
                 else {
                     const elementContextualType = getContextualTypeForElementExpression(contextualType, index);
                     const type = checkExpressionForMutableLocation(e, checkMode, elementContextualType);
-                    elementTypes.push(type);
+                    elementTypes.push(getWidenedTypeOfUniqueESSymbolType(type));

```

Comment:
You can snub out all unique symbol types at the source by checking the syntactic context in `checkIdentifier` and `checkPropertyAccess`.

Reply:
I have started looking at this, but I'm beginning to think that is too aggressive. Consider this example:

```ts
const x = Symbol();
const y = Symbol();

const obj = {
  [x]: number;
  [y]: string;
}

const a = obj[Math.random() * 2 ? x : y];
```

You would expect `a` to have the type `number | string`. I think we only want to widen in a few cases:

- Assignments
- Initializers (when the declaration does not have a type annotation)
- `return`
- `yield`
- Array literal expressions
- Inference

Trying to widen at an identifier, property access, or element access based on the parent ended up being far more complicated. 

I've added a number of tests cases for when you would expect widening to occur that I will push shortly.